bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/util/ThingHelper.java

```diff
@@ -90,10 +95,28 @@ private static String toString(List<Channel> channels) {
     }
 
     public static void addChannelsToThing(Thing thing, Collection<Channel> channels) {
+        List<Channel> copy = new LinkedList<>(thing.getChannels());
+        copy.addAll(channels);
+        ensureUnique(copy);
         ((ThingImpl) thing).getChannelsMutable().addAll(channels);
     }
 
     /**
+     * Ensures that there are no duplicate channels in the list (i.e. not using the same ChannelUID)
+     *
+     * @param channels the {@link List} of channels to check
+     * @throws IllegalArgumentException in case there are duplicate channels found
+     */
+    public static void ensureUnique(List<Channel> channels) {
```

Comment:
Thanks @maggu2810, I took the freedom to include your suggestions here (and adding an also-by line to the commit). Hope that's okay?